### ISSUE TITLE
CLI broken in ES modules due to `cosmiconfig-typescript-loader`

### ISSUE THREAD

**Issue by Diizzayy** [dotansimha/graphql-code-generator#8159]
### Describe the bug

The programmatic CLI usage in ES modules appears to be broken, this might be due in part to kamilkisiela/graphql-config#1069 by @Codex- as the error occurs with `cosmiconfig-typescript-loader/dist/esm/index.js` which is the package of subject in the aforementioned PR.

It's also worth mentioning that simply importing anything from `@graphql-codegen/cli` in an ES module triggers this crash.

@dotansimha @B2o5T Any insight here would be highly appreciated, This is a roadblock for me.

### Error

```sh
(node:70003) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
/home/diizzayy/dev/gqlbug/node_modules/cosmiconfig-typescript-loader/dist/esm/index.js:1
import { TypeScriptLoader } from "./loader";
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1033:15)
    at Module._compile (node:internal/modules/cjs/loader:1069:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:170:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:198:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:385:24)
error Command failed with exit code 1.
```

### Your Example Website or App

https://github.com/Diizzayy/codegen-es-bug

### Steps to Reproduce the Bug or Issue

I've produced a basic reproduction

simply run `yarn install` followed by `yarn start`

### Expected behavior

the graphql-codegen CLI should properly generate a file called `gql.ts`

### Platform

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.15.1`
- Package Manager:  `yarn@1.22.18`
- `@graphql-codegen/cli`:          `^2.11.2`
- `@graphql-codegen/typescript`:      `^2.7.2`
- `graphql`:   `^16.5.0`
------------------------------

**Comment by Codex-** [comment]
taking a look into this now, a bit painful how the resolution doesn't seem to be being respected by node.

**Comment by Codex-** [comment]
I have fixed this locally, preparing a release currently:
```ascii
➜  codegen-es-bug git:(main) ✗ yarn start
yarn run v1.22.19
$ node --trace-warnings index.mjs
✔ Parse Configuration
❯ Generate outputs
  ❯ Generate gql.ts
    ⠋ Load GraphQL schemas
    ◼ Load GraphQL documents
    ◼ Generate
(node:4250) ExperimentalWarning: stream/web is an experimental feature. This feature could change at any time
    at emitExperimentalWarning (node:internal/util:224:11)
    at node:stream/web:7:1
    at NativeModule.compileForInternalLoader (node:internal/bootstrap/loaders:312:7)
    at NativeModule.compileForPublicLoader (node:internal/bootstrap/loaders:252:10)
✔ Parse Configuration
✔ Generate outputs
Done in 3.12s.
```

**Comment by Diizzayy** [comment]
@Codex- Wonderful! Thank you for addressing this so promptly.

**Comment by Codex-** [comment]
@Diizzayy this will require a bump from the graphql-config authors: https://github.com/kamilkisiela/graphql-config/pull/1096

Hopefully this is put through overnight (it's 1am my time 😅)

**Comment by Diizzayy** [comment]
@Codex- I'm currently making the PR as we speak

**Comment by Codex-** [comment]
I have already made the PR on graphql-config 😄

**Comment by B2o5T** [comment]
released in [graphql-config@4.3.3](https://github.com/kamilkisiela/graphql-config/releases/tag/v4.3.3)

**Comment by Diizzayy** [comment]
@B2o5T @Codex-  I can confirm it's been resolved by your changes.
